src/components/Table/index.jsx

```diff
@@ -116,7 +117,7 @@ const Table = ({
       columnData
     );
 
-  const sortedColumns = preserveTableStateInQuery
+  const sortedColumns = isDefaultPageChangeHandler
     ? setSortFromURL(curatedColumnsData)
     : curatedColumnsData;
```

Comment:
curatedColumns data is derived from queryParams which is obtained by calling `const queryParams = getQueryParams();`. But getQueryParams is just a utility function. It won't listen to URL changes and rerender the component on its own. Prefer `useQueryParams` when you are expecting your component to re-render in response to URL query params changes.
I believe I have already conveyed this message to you.

Even though this change was done earlier by someone else and you are making minor changes on top of it, the scope of the current issue is to optimize everything for future use. So, it is your responsibility to clean any already existing suboptimal code. Go through the whole codebase & see if anything related to pagination/query params in this component is suboptimal. Clean them up.